G/Geant4/bundled/patches/windows.patch

```diff
@@ -0,0 +1,118 @@
+diff --git a/source/externals/ptl/src/Threading.cc b/source/externals/ptl/src/Threading.cc
+index 324d6c12d7..a7553d8db2 100644
+--- a/source/externals/ptl/src/Threading.cc
++++ b/source/externals/ptl/src/Threading.cc
+@@ -27,7 +27,7 @@
+ #include "PTL/Utility.hh"
+ 
+ #if defined(PTL_WINDOWS)
+-#    include <Windows.h>
++#    include <windows.h>
+ #endif
+ 
+ #if defined(PTL_MACOS)
+diff --git a/source/global/management/src/G4FindDataDir.cc b/source/global/management/src/G4FindDataDir.cc
+index 8c80180f71..7c5a6e0158 100644
+--- a/source/global/management/src/G4FindDataDir.cc
++++ b/source/global/management/src/G4FindDataDir.cc
+@@ -34,7 +34,7 @@
+ #include <cstdlib>
+ #include <cstring>
+ 
+-#if defined(_MSC_VER)
++#if defined(_WIN32)
+ #define setenv(name, value, overwrite) _putenv_s(name, value)
+ #endif
+ 
+diff --git a/source/global/management/src/G4Threading.cc b/source/global/management/src/G4Threading.cc
+index 3c6bd77b14..fe09e672c7 100644
+--- a/source/global/management/src/G4Threading.cc
++++ b/source/global/management/src/G4Threading.cc
+@@ -35,7 +35,7 @@
+ #include "globals.hh"
+ 
+ #if defined(WIN32) || defined(__MINGW32__)
+-#  include <Windows.h>
++#  include <windows.h>
+ #else
+ #  include <sys/syscall.h>
+ #  include <sys/types.h>
+diff --git a/source/processes/electromagnetic/dna/utils/include/G4MoleculeGun.hh b/source/processes/electromagnetic/dna/utils/include/G4MoleculeGun.hh
+index ef764260ac..54f9a8a17f 100644
+--- a/source/processes/electromagnetic/dna/utils/include/G4MoleculeGun.hh
++++ b/source/processes/electromagnetic/dna/utils/include/G4MoleculeGun.hh
+@@ -110,6 +110,8 @@ protected:
+   void ShootAtFixedPosition(G4MoleculeGun*){}
+ };
+ 
++template<> void TG4MoleculeShoot<G4Track>::Shoot(G4MoleculeGun* gun);
++
+ template<typename TYPE>
+ G4shared_ptr<G4MoleculeShoot> G4MoleculeShoot::ChangeType()
+ {
+diff --git a/source/processes/hadronic/models/lend/include/MCGIDI.h b/source/processes/hadronic/models/lend/include/MCGIDI.h
+index e11303d2f4..3e4fa17887 100644
+--- a/source/processes/hadronic/models/lend/include/MCGIDI.h
++++ b/source/processes/hadronic/models/lend/include/MCGIDI.h
+@@ -9,7 +9,7 @@
+ #define MCGIDI_VERSION_MINOR 0
+ #define MCGIDI_VERSION_PATCHLEVEL 0
+ 
+-#ifdef WIN32
++#ifdef _WIN32
+ #define M_PI 3.141592653589793238463
+ #endif
+ 
+diff --git a/source/processes/hadronic/models/lend/include/nf_specialFunctions.h b/source/processes/hadronic/models/lend/include/nf_specialFunctions.h
+index 4321f47a59..fe65149e68 100644
+--- a/source/processes/hadronic/models/lend/include/nf_specialFunctions.h
++++ b/source/processes/hadronic/models/lend/include/nf_specialFunctions.h
+@@ -11,7 +11,7 @@
+ 
+ #include "nf_utilities.h"
+ 
+-#ifdef WIN32
++#ifdef _WIN32
+ #define M_PI 3.141592653589793238463
+ #endif
```

Comment:
Side note, the way to get `M_PI` on Windows is to define `_USE_MATH_DEFINES` before loading `math.h`/`cmath`: https://learn.microsoft.com/en-us/cpp/c-runtime-library/math-constants?view=msvc-170

Reply:
Yes I know, I used in one occasion.  But I wanted to do minimal patch. They define this several times.